chaos_genius/alerts/anomaly_alerts.py

```diff
@@ -171,56 +251,33 @@ def _find_point(self, point, prev_data):
                 break
         return intended_point
 
-    def _save_nl_message_daily_freq(self, anomaly_data: List[dict], kpi: Kpi):
-        """Saves change message for every point, for a daily frequency KPI."""
-        time_diff = datetime.timedelta(days=1, hours=0, minutes=0)
-
-        # TODO: fix circular import
-        from chaos_genius.controllers.digest_controller import get_previous_data
-
-        prev_day_data = get_previous_data(kpi.id, self.anomaly_end_date, time_diff)
+    def _save_nl_message(self, anomaly_data: List[dict]):
+        """Constructs and saves change message for every point."""
+        kpi = get_kpi_obj(self.kpi_id)
+        if kpi is None:
+            for point in anomaly_data:
+                point["nl_message"] = "KPI does not exist"
 
-        prev_day_data = [anomaly_point.as_dict for anomaly_point in prev_day_data]
+            logger.info(f"KPI ID - {self.kpi_id} does not exist")
```

Comment:
Would be better to have this as a warning and include both KPI and Alert ID (and in the same format as the other logs)
```suggestion
            logger.warn(f"(KPI: {self.kpi_id}, Alert: {self.alert_id}) KPI does not exist")
```